src/test/regress/bin/diff

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```

Comment:
> Is there a way to not error out if bash is not installed /bin/bash but somewhere else? In that case, we could at least fail only few tests.
> Also, I'm not very experienced with this, but some search tells that it is more portable to use #!/usr/bin/env bash, do you have a better idea?

Changed to `#!/usr/bin/env bash`

> To make this more robust/portable,  for example, would it be possible to print a warning and continue with the actual diff -- and skipping sed -- in case any error happens at any step? I mean, can we have a something like a try/catch? I'd be a lot more comfortable to merge with that.

I am not sure what would be the effect of printing warnings (or any extra information), as in I don't know how will `pg_regress` will treat them. That needs more testing/code reading before we do that.

For error checking and falling back to `diff`, I'd be more comfortable when we see the first case that current code doesn't work, so _we make sure that our fallback code actually works_.